### PR TITLE
workspace: Update qdldl and osqp to latest release

### PR DESCRIPTION
**Pull request by jwnimmer-tri**
Relates #11883.

---
This change is [Reviewable](https://reviewable.io/reviews/robotlocomotion/drake/12047)

### CODE REVIEW

**Comment by jwnimmer-tri** [comment]
I also would like +@jamiesnape for feature review of the build changes.  (I still need to fix up the README though.)

**Comment by jwnimmer-tri** [comment]
> (I still need to fix up the README though.)

Done.

File: tools/workspace/qdldl/README.md [hunk 1]
```diff
@@ -0,0 +1,13 @@
+Within drake, QDLDL is used by OSQP.
```
Comment (jamiesnape) [inline tools/workspace/qdldl/README.md]: drake -> Drake

File: tools/workspace/qdldl/README.md [hunk 2]
```diff
@@ -0,0 +1,13 @@
...
+
+When changing the release of either QDLDL or OSQP used by Drake, ideally try to
+keep Drake's pin of QDLDL aligned with what Drake's pin of OSQP desires.
+
+For example, if Drake pinned OSQP to commit = "v0.4.1" then look at
+https://github.com/oxfordcontrol/osqp/tree/v0.4.1/lin_sys/direct/qdldl
+which shows that OSQP 0.4.1 prefers a pin of 7ab0fca for QDLDL.
+
+Looking at  https://github.com/oxfordcontrol/qdldl/commit/7ab0fca we
```
Comment (jamiesnape) [inline tools/workspace/qdldl/README.md]: remove extra space

File: tools/workspace/qdldl/README.md [hunk 3]
```diff
@@ -0,0 +1,13 @@
...
+see that it was approximately the 0.1.3 release.  (The actual v0.1.3
+release tag on QDLDL is on the merge commit of 7ab0fca onto master.)
+So, Drake shoul ideally pin QDLDL commit = "v0.1.3".
```
Comment (jamiesnape) [inline tools/workspace/qdldl/README.md]: should -> should

**Comment by jwnimmer-tri** [comment]
@sherm1 this is ready for platform review now.  To review the README you'll have to view the latest copy on GitHub (https://github.com/jwnimmer-tri/drake/blob/new-release-osqp/tools/workspace/qdldl/README.md) since Reviewable hasn't picked it up yet, but everything else is accurate in r1 of Reviewable (it has not changed since the original push).

**Comment by jwnimmer-tri** [comment]
@jamiesnape Please check the wording of the large new API comment, to see if you have any issues with it.

**Comment by jamiesnape** [comment]
It's fine.

**Pull request merged**